src/main/java/leets/memoriessquare/domain/photo/presentation/PhotoController.java

```diff
@@ -0,0 +1,30 @@
+package leets.memoriessquare.domain.photo.presentation;
+
+import leets.memoriessquare.domain.photo.presentation.dto.PhotoDTO;
+import leets.memoriessquare.domain.photo.presentation.dto.UploadPhotoResponse;
+import leets.memoriessquare.domain.photo.presentation.dto.UploadPhotoRequest;
+import leets.memoriessquare.domain.photo.usecase.UploadPhoto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import jakarta.validation.Valid;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/photo")
+@RequiredArgsConstructor
+public class PhotoController {
+    private final UploadPhoto uploadPhoto;
+
+    @PostMapping("/upload")
+    public UploadPhotoResponse uploadPhoto(
+            @Valid UploadPhotoRequest request,
+            @RequestParam("file") MultipartFile file,
+            @AuthenticationPrincipal(expression = "userId") UUID userId) throws Exception {
```

Comment:
```suggestion
            @AuthenticationPrincipal OAuthDetails auth) throws Exception {
```
userId(UUID)를 가져오기 위해서는 `OAuthDetails` 클래스를 받아서 `auth.getId()` 형태로 가져와야 합니다.

(import 하는 것때문에 suggestion 적용하지 말고 직접 수정하는 게 나을거에요)